components/eam/src/physics/cam/zm_conv.F90

```diff
@@ -540,6 +659,85 @@ subroutine zm_convr(lchnk   ,ncol    , &
    heat(:,:) = 0._r8
    mcon(:,:) = 0._r8
    rliq(:ncol)   = 0._r8
+   rice(:ncol)   = 0._r8    
+
+   if (zm_microp) then
```

Comment:
It would be nice to move all these initializations to a routine in "zm_microphysics" rather than add so many more lines to this already huge file.

Reply:
All allocations and initializations are moved into corresponding subroutines.